### PR TITLE
fix(timeline): single event row issue fixed

### LINKED ISSUE #2676: MiniGantt extent breaks scrolling when only one event exists

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #2676

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

File: packages/timeline/src/MiniGantt.ts [hunk 1]
```diff
@@ -75,10 +75,15 @@ export class MiniGantt extends SVGWidget {
...
     extent() {
-        if (this.rootExtent) {
-            return [this.rootExtent[1], this.rootExtent[2]];
+        const extent = this.rootExtent ? [this.rootExtent[1], this.rootExtent[2]] : this.fullExtent();
+        if (extent[0] === extent[1]) {
+            const parser = d3TimeParse(this.timePattern());
+            const formatter = d3TimeFormat(this.timePattern());
+            const dt = parser(extent[0]);
+            extent[0] = formatter(new Date(dt.setFullYear(dt.getFullYear() - 1)));
+            extent[1] = formatter(new Date(dt.setFullYear(dt.getFullYear() + 2)));
```
Comment (GordonSmith) [inline packages/timeline/src/MiniGantt.ts]: Shouldn't this be:
```
const dtHi = parser(extent[1]);
extent[1] = formatter(new Date(dtHi.setFullYear(dtHi.getFullYear() +1)));
```

**Comment by GordonSmith** [comment]
@jbrundage Travis error (legit).

File: packages/timeline/src/MiniGantt.ts (hunk 1 above)
Comment (jbrundage) [reply packages/timeline/src/MiniGantt.ts]: That conditional (`extent[0] === extent[1]`) evaluates to true when there's only one unique timestamp in the dataset.

**Comment by jbrundage** [comment]
@GordonSmith the Travis error will go away once the pending MiniGantt enhancements are merged

**Comment by jbrundage** [comment]
These changes were merged with: https://github.com/hpcc-systems/Visualization/commit/3a7375df7ca8262b99702cc3af65e7b0d2dd636f

**Pull request closed without merging**